opm/autodiff/FullyImplicitBlackoilSolver.hpp

```diff
@@ -194,6 +194,12 @@ namespace Opm {
         std::vector<int>         primalVariable_;
 
         // Private methods.
+
+        // return true if wells are available
+        bool wellsActive() const { return wells_ ? wells_->number_of_wells > 0 : false ; }
+        // return wells object
+        const Wells& wells () const { assert( wells_ ); return *wells_; }
```

Comment:
Are you sure? The webpage http://www.cplusplus.com/reference/cassert/assert/ gives
```
void assert (int expression);
```
as the form of the macro, but I guess it may be specified differently in the standard? If this really is illegal, I fear I have done a few assert(ptr) errors over the years... Of course there is no harm in using your suggested alternatives anyway.